### PR TITLE
fix: adapt content manager to use new doc service api

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Adapt content manager to use new document service format.

We had an entity-manager service before that was handling single types and collection types at the same time. this splitted that service into to, similarly as what we have in the doc service

### CODE REVIEW

File: packages/core/content-manager/server/src/services/uid.ts [hunk 1]
```diff
@@ -53,36 +53,51 @@ export default ({ strapi }: { strapi: Strapi }) => ({
...
+      foundDocuments = await strapi.documents<Schema.CollectionType>(contentTypeUID).findMany({
+        filters: {
+          [field]: value,
+        },
+        locale: locale ?? null,
+        // TODO: Check UX. When modifying an entry, it only makes sense to check for collisions with other drafts
+        // However, when publishing this "available" UID might collide with another published entry
+        status: 'draft',
+      });
+    }
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/services/uid.ts:79]: These are the type of cases that could add a bit more friction cc @alexandrebodin

File: packages/core/content-manager/server/src/controllers/single-types.ts [hunk 2]
```diff
@@ -281,19 +280,21 @@ export default {
...
   async countDraftRelations(ctx: any) {
     const { userAbility } = ctx.state;
     const { model } = ctx.params;
-
-    const entityManager = getService('entity-manager');
+    const { query } = ctx.request;
+    const singleTypes = getService('single-types');
     const permissionChecker = getService('permission-checker').create({ userAbility, model });
 
+    const { locale } = getDocumentDimensions(query);
```
Comment (alexandrebodin) [inline packages/core/content-manager/server/src/controllers/single-types.ts:296]: Not sure we want that generic dimensions term as of now :)

File: packages/core/content-manager/server/src/controllers/single-types.ts [hunk 3]
```diff
@@ -281,19 +280,21 @@ export default {
     }
 
     ctx.body = await pipeAsync(
-      (document) => entityManager.discard(document, model, { locale }),
+      () => singleTypes.discard(model, { locale }),
```
Comment (alexandrebodin) [inline packages/core/content-manager/server/src/controllers/single-types.ts:283]: should we harmonize names (discardDraft) ?

File: packages/core/content-manager/server/src/controllers/single-types.ts [hunk 4]
```diff
@@ -73,8 +72,8 @@ const createOrUpdateDocument = async (ctx: any, opts?: { populate: object }) =>
   }
 
   const sanitizedBody = await sanitizeFn(body);
-  return entityManager.update(documentVersion || { id: otherDocumentVersion.documentId }, model, {
-    data: sanitizedBody,
+  return singleTypes.createOrUpdate(model, {
```
Comment (alexandrebodin) [inline packages/core/content-manager/server/src/controllers/single-types.ts:75]: we probably can rename it update as it's know clearly 2 different APIs like in the doc service for consistency

File: packages/core/content-manager/server/src/services/collection-types.ts [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+  const modelDef = strapi.getModel(uid);
+  const sanitizedDocument = await sanitize.sanitizers.defaultSanitizeOutput(modelDef, document);
+
+  strapi.eventHub.emit(event, {
+    model: modelDef.modelName,
+    entry: sanitizedDocument,
+  });
+};
+
+const buildDeepPopulate = (uid: Common.UID.CollectionType) => {
```
Comment (alexandrebodin) [inline packages/core/content-manager/server/src/services/collection-types.ts:33]: could this be shared code between single & collection types ?

File: packages/core/content-manager/server/src/services/uid.ts (hunk 1 above)
Comment (alexandrebodin) [reply packages/core/content-manager/server/src/services/uid.ts:79]: Nice usecase to consider indeed 👍

File: packages/core/core/src/services/document-service/document-engine.ts [hunk 6]
```diff
@@ -313,7 +313,7 @@ const createDocumentEngine = ({
     return this.delete(uid, documentId, {
       ...params,
       lookup: { ...params?.lookup, publishedAt: { $ne: null } },
-    }) as any;
+    }).then(({ deletedEntries }) => ({ versions: deletedEntries })) as any;
```
Comment (alexandrebodin) [inline packages/core/core/src/services/document-service/document-engine.ts:316]: that makes me think 🤔  unpublishing can never overwrite the draft  before deletion for now ? maybe we will want to allow it later.

File: packages/core/content-manager/server/src/controllers/single-types.ts (hunk 2 above)
Comment (Marc-Roig) [reply packages/core/content-manager/server/src/controllers/single-types.ts:296]: sure! let me change it

File: packages/core/core/src/services/document-service/document-engine.ts (hunk 6 above)
Comment (Marc-Roig) [reply packages/core/core/src/services/document-service/document-engine.ts:316]: not directly, and we do have that exact same usecase on the content manager. We are discarding the draft + unpublishing

**Pull request merged**